internal/wasm/host.go

```diff
@@ -176,6 +176,26 @@ func (m *Module) buildHostFunctions(moduleName string) (functions []*FunctionIns
 			Index:  Index(idx),
 		}
 		f.DebugName = wasmdebug.FuncName(moduleName, functionNames[f.Index].Name, f.Index)
+		if functionListenerFactory != nil {
+			params := make([]ValueInfo, len(f.ParamTypes()))
+			for i, p := range f.ParamTypes() {
+				params[i].Type = p
+			}
+			results := make([]ValueInfo, len(f.ResultTypes()))
+			for i, r := range f.ResultTypes() {
+				results[i].Type = r
+			}
+
+			// Cannot access parameter names for host functions.
```

Comment:
ps I plan to change the ModuleBuilder to supply param names as often these should match something else, like wasi definitions